packages/react-storage/src/components/StorageBrowser/storageBrowser/types.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  ActionConfigs,
+  DefaultActionConfigs,
+  DefaultActionKeys,
+} from '../actions/configs';
+import { createUseAction } from '../actions/createUseAction';
+import { StorageBrowserElements } from '../context/elements';
+import { Config } from '../createProvider';
+
+interface CreateStorageBrowserInput<
+  DefinedActionConfigs extends ActionConfigs,
+> {
+  config: Config;
+  elements?: Partial<StorageBrowserElements>;
+  actions?: DefinedActionConfigs;
+}
+
+interface CreateStorageBrowserOutput<
+  DefinedActionConfigs extends ActionConfigs,
+> {
+  StorageBrowser: {
+    (props: {
+      type: ExtractComponentNamesFromActionConfigs<DefinedActionConfigs>;
+    }): React.JSX.Element;
+    displayName: string;
+    Provider: (props: { children?: React.ReactNode }) => React.JSX.Element;
+  } & CreateActionDerivedViews<DefinedActionConfigs>; // & the action derived views components
```

Comment:
Is " // & the action derived views components` noting a todo or is it meant as a comment?

Reply:
This is really for readability to help who ever looking at the type programming code to understand what's going on. 😅

Reply:
Cool cool, maybe we place it on the line above and remove the `&`? Read it as a todo to add an additional intersection operator 😅

Reply:
The formatter replaces the comment for me, ha! I've actually remove this comment. After refactoring earlier the purpose of this `&` became more obvious.